addons/payment/models/ir_ui_view.py

```diff
@@ -9,8 +9,8 @@ class IrUiView(models.Model):
 
     @api.ondelete(at_uninstall=False)
     def _unlink_if_not_referenced_by_acquirer(self):
-        referencing_acquirers = self.env['payment.acquirer'].search([
+        referencing_acquirers_sudo = self.env['payment.acquirer'].sudo().search([
             '|', ('redirect_form_view_id', 'in', self.ids), ('inline_form_view_id', 'in', self.ids)
-        ])
-        if referencing_acquirers:
```

Comment:
Thanks for the fix!

Out of curiosity, isn't this method a convoluted way of setting `ondelete="restrict"` on `redirect_form_view_id` and `inline_form_view_id`, which would guarantee FK integrity?

Deleting acquirer views should be rather unlikely, the generic error of `ondelete="restrict"` might have been good enough, no?

Reply:
Yes, I agree. I guess I got a bit too carried away when refactoring the module and two `ondelete`'s would have been enough. I will change that in master.